CMakeLists.txt

```diff
@@ -64,31 +60,31 @@ include(check_packages)
 include(swig)
 include(coveralls)
 
-# add PaddlePaddle version
-if(DEFINED ENV{PADDLE_VERSION})
-    add_definitions(-DPADDLE_VERSION=\"$ENV{PADDLE_VERSION}\")
+# Set PaddlePaddle version to Git tag name or Git commit ID.
+find_package(Git REQUIRED)
+execute_process(
+  COMMAND ${GIT_EXECUTABLE} describe --tags --exact-match --abbrev=0
+  WORKING_DIRECTORY ${PROJ_ROOT}
+  OUTPUT_VARIABLE GIT_TAG_NAME
+  RESULT_VARIABLE GIT_RESULT
+  ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(NOT ${GIT_RESULT})
+  set(PADDLE_VERSION ${GIT_TAG_NAME})
```

Comment:
If only use git tag name, the paddle version will be "wrong" for developers trying to build directly from master, or experimenting himself.
Last time I built tensorflow master, I got version as `0.11.head`, which I think is in format `tag.commit`.